### PR TITLE
Feature: Braintree PayPal v6 Provider

### PR DESCRIPTION
**Pull request by EvanReinstein**
Adds a V6 `BraintreePayPalProvider` and `useBraintreePayPal` hook that coordinates with the Braintree PayPal Checkout V6 module. The provider validates the Braintree namespace, creates a client and checkout instance, loads the PayPal SDK, and exposes the checkout instance to child components via context.                                                                                            
                                                                                                                                       
  - `BraintreePayPalProvider`: accepts a `namespace` (the Braintree window object) and `braintreeClientToken`, manages initialization lifecycle with proper cleanup on unmount                                                                                             
  - `useBraintreePayPal`: hook to consume the provider's context (checkout instance, loading status, error, hydration state)
  - Fully typed `BraintreePayPalCheckoutInstance` interface matching the [Braintree Web SDK docs](https://braintree.github.io/braintree-web/current/PayPalCheckoutV6.html), including typed options for all session/payment methods
  - SSR support via `isHydrated` flag, consistent with the existing `PayPalProvider`
                                                                                                                                       
  Also includes:                                                                                                                       
  - Rename `PayPalProviderEnums.ts` to `ProviderEnums.ts` and consolidate Braintree dispatch actions there. This ended up touching a bunch of files, sorry!
  - Suppress pre-existing `console.error` noise in `useEligibleMethods` tests via `useErrorTestUtil`

### CODE REVIEW

File: packages/react-paypal-js/src/v6/hooks/useEligibleMethods.test.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import React from "react";
 import { renderHook, act } from "@testing-library/react-hooks";
 
+import "./useErrorTestUtil";
```
Comment (EvanReinstein) [inline packages/react-paypal-js/src/v6/hooks/useEligibleMethods.test.ts:4]: This fixes a console.error log that logs innocuously during tests.

File: packages/react-paypal-js/src/v6/hooks/useEligibleMethods.test.ts [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
+
 import { useEligibleMethods } from "./useEligibleMethods";
 import { PayPalContext } from "../context/PayPalProviderContext";
 import { PayPalDispatchContext } from "../context/PayPalDispatchContext";
 import {
     INSTANCE_DISPATCH_ACTION,
     INSTANCE_LOADING_STATE,
-} from "../types/PayPalProviderEnums";
+} from "../types/ProviderEnums";
```
Comment (EvanReinstein) [inline packages/react-paypal-js/src/v6/hooks/useEligibleMethods.test.ts:12]: This file name change is affecting a lot of files 🙃

File: packages/react-paypal-js/src/v6/types/braintree.ts
Comment (EvanReinstein) [inline packages/react-paypal-js/src/v6/types/braintree.ts:1]: The majority of the types were pulled from [these BT docs](https://braintree.github.io/braintree-web/current/PayPalCheckoutV6.html).

File: packages/react-paypal-js/src/v6/components/Braintree/BraintreePayPalProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+                        value: toError(error),
+                    });
+                }
+            }
+        };
+
+        initialize();
+
+        return () => {
+            isSubscribed = false;
```
Comment (HackTheW2d) [inline packages/react-paypal-js/src/v6/components/Braintree/BraintreePayPalProvider.tsx:159]: Do we want to add teardown() method somewhere?

Refer: https://braintree.github.io/braintree-web/current/PayPalCheckoutV6.html#teardown
In my understanding, `isSubscribed` prevents updating on braintree side. `teardown`  would release resources from braintree. I am not sure if there'll be leakage if token has changed since we didn't release resources in React state once the first checkoutInstance is dispatched. But this is an edge case and I might be wrong on this.

Comment (EvanReinstein) [reply packages/react-paypal-js/src/v6/components/Braintree/BraintreePayPalProvider.tsx:159]: `isSubscribed` does more than just prevent updates on the BT side, it also works to prevent React state updates or dispatches.

I have updated this to include the `teardown` usage, we can test it out once the UI component is available to really see if this has a negative impact.

File: packages/react-paypal-js/src/v6/types/braintree.ts [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+}
+
+export interface BraintreeLoadPayPalSDKOptions {
+    env?: string;
+}
+
+// ---- Session object ----
+
+export interface BraintreePaymentSession {
+    start: () => void;
```
Comment (HackTheW2d) [inline packages/react-paypal-js/src/v6/types/braintree.ts:300]: It seems like in the example, there's `resume()` and `hasReturned()` methods as well in braintree's `createOneTimePaymentSession`: https://braintree.github.io/braintree-web/current/PayPalCheckoutV6.html#createOneTimePaymentSession-examples

Not sure if this is the case for all `BraintreePaymentSession`. Just want to give a heads up here :)

Comment (EvanReinstein) [reply packages/react-paypal-js/src/v6/types/braintree.ts:300]: Yea, good point. I think that we'll enhance this interface to potentially be the base interface and provide unique types for each session returned. 

The hooks themselves will handle the `resume` and `hasReturned` values in order to facilitate the direct app switch.

File: packages/react-paypal-js/src/v6/hooks/useBraintreePayPal.ts
Comment (HackTheW2d) [inline packages/react-paypal-js/src/v6/hooks/useBraintreePayPal.ts:1]: Are we making this file `.tsx` for some intent? Didn't see any tsx syntax here :)

**Review by HackTheW2d — APPROVED** [review]
Great work on adding this new provider!

File: packages/react-paypal-js/src/v6/hooks/useBraintreePayPal.ts
Comment (EvanReinstein) [reply packages/react-paypal-js/src/v6/hooks/useBraintreePayPal.ts:1]: Oh you know what, that was my mistake when I created a handful of files to scaffold the structure

**Review by nityasp — COMMENTED** [review]
I guess `BraintreeProvider` doesn't need auto-memorization like `PayPalProvider` right? because the namespace is a stable reference and its loaded once globally. Just asking to clear my understanding.

**Review by nityasp — APPROVED** [review]
The `BraintreeProvider` is clean and on point. Thanks for working on this @EvanReinstein
